### PR TITLE
Add XRC handlers for wxInfoBar

### PR DESCRIPTION
**Pull request by thesiv**

### CODE REVIEW

File: docs/doxygen/overviews/xrc_format.h [hunk 1]
```diff
@@ -1221,6 +1231,36 @@ page.
...
+    <showeffect>wxSHOW_EFFECT_EXPAND</showeffect>
+    <hideeffect>wxSHOW_EFFECT_SLIDE_TO_RIGHT</hideeffect>
+    <object class="button" name="wxID_UNDO"/>
+    <object class="button" name="wxID_REDO">
+        <label>Redo Custom Label</label>
+    </object>
+</object>
+@endcode
+
+
```
Comment (vadz) [inline docs/doxygen/overviews/xrc_format.h]: We need `@since 3.1.3` here.

File: docs/doxygen/overviews/xrc_format.h [hunk 2]
```diff
@@ -1221,6 +1231,36 @@ page.
 @endTable
 
 
+@subsubsection xrc_wxinfobar wxInfoBar
+
```
Comment (vadz) [inline docs/doxygen/overviews/xrc_format.h]: I'd either mention that all the sub-elements are optional or explicitly add "Optional" and/or "default" to each element description, to make it clear that they can all be omitted.

File: src/xrc/xh_infobar.cpp [hunk 3]
```diff
@@ -0,0 +1,111 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        src/xrc/xh_infobar.cpp
+// Purpose:     XML resource handler for wxInfoBar
+// Author:      Ilya Sinitsyn
+// Created:     2019/09/25
```
Comment (vadz) [inline src/xrc/xh_infobar.cpp]: Very minor, but let's use ISO format:
```suggestion
// Created:     2019-09-25
```

File: src/xrc/xh_infobar.cpp [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    XRC_ADD_SHOW_EFFECT(wxSHOW_EFFECT_SLIDE_TO_RIGHT);
+    XRC_ADD_SHOW_EFFECT(wxSHOW_EFFECT_SLIDE_TO_TOP);
+    XRC_ADD_SHOW_EFFECT(wxSHOW_EFFECT_SLIDE_TO_BOTTOM);
+    XRC_ADD_SHOW_EFFECT(wxSHOW_EFFECT_BLEND);
+    XRC_ADD_SHOW_EFFECT(wxSHOW_EFFECT_EXPAND);
+}
+
+wxObject *wxInfoBarXmlHandler::DoCreateResource()
+{
+    if( m_class == wxT("wxInfoBar") )
```
Comment (vadz) [inline src/xrc/xh_infobar.cpp]:
```suggestion
    if ( m_class == "wxInfoBar" )
```
(alternatively, use `wxS()`, if you really want to, but avoid `wxT()` in any new code)

File: src/xrc/xh_infobar.cpp [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+        m_insideBar = true;
+        CreateChildrenPrivately(control);
+        m_insideBar = false;
+
+        return control;
+    }
+    else
+    {
+        // on the inside now.
```
Comment (vadz) [inline src/xrc/xh_infobar.cpp]:
```suggestion
        // inside the element now,
```

**Review by vadz — COMMENTED** [review]
Very nice, thanks!

Could you please just fix the few minor things below?

**Comment by thesiv** [comment]
Fixed.

**Pull request closed without merging**